src/libraries/Microsoft.PowerFx.Core/Public/Config/Features.cs

```diff
@@ -2,77 +2,125 @@
-        DisableRowScopeDisambiguationSyntax = 0x4,
+        internal bool DisableRowScopeDisambiguationSyntax { get; set; }
 
         /// <summary>
-        /// Enable Identifier support for describing column names
-        /// Added on 6th December 2022.
+        /// Enable Identifier support for describing column names.
         /// </summary>
-        SupportColumnNamesAsIdentifiers = 0x8,        
+        internal bool SupportColumnNamesAsIdentifiers { get; set; }        
 
         /// <summary>
         /// Enforces strong-typing for builtin enums, rather than treating
-        /// them as aliases for values of string/number/boolean types
-        /// Added March 2023.
+        /// them as aliases for values of string/number/boolean types.
         /// </summary>
-        StronglyTypedBuiltinEnums = 0x10,
+        internal bool StronglyTypedBuiltinEnums { get; set; }
 
         /// <summary>
         /// Updates the IsEmpty function to only allow table arguments, since it
         /// does not work properly with other types of arguments.
         /// </summary>
-        RestrictedIsEmptyArguments = 0x20,
+        internal bool RestrictedIsEmptyArguments { get; set; }
 
         /// <summary>
         /// Allow delegation for async calls (delegate using awaited call result).
-        /// Added March 2023.
         /// </summary>
-        AllowAsyncDelegation = 0x40,
+        internal bool AllowAsyncDelegation { get; set; }
 
         /// <summary>
         /// Allow delegation for impure nodes.
-        /// Added March 2023.
         /// </summary>
-        AllowImpureNodeDelegation = 0x80,
+        internal bool AllowImpureNodeDelegation { get; set; }
 
         /// <summary>
         /// Updates the FirstN/LastN functions to require a second argument, instead of
         /// defaulting to 1.
         /// </summary>
-        FirstLastNRequiresSecondArguments = 0x100,
+        internal bool FirstLastNRequiresSecondArguments { get; set; }
+
+        internal bool PowerFxV1CompatibilityRules { get; set; }
+
+        internal static Features None => new Features();
 
-        /// <summary>        
+        public static Features PowerFxV1 => new Features
+        {
+            TableSyntaxDoesntWrapRecords = true,
+            ConsistentOneColumnTableResult = true,
+            DisableRowScopeDisambiguationSyntax = true,
+            SupportColumnNamesAsIdentifiers = true,
+            StronglyTypedBuiltinEnums = true,
+            RestrictedIsEmptyArguments = true,
+            AllowAsyncDelegation = true,
+            AllowImpureNodeDelegation = true,
+            FirstLastNRequiresSecondArguments = true,
+            PowerFxV1CompatibilityRules = true,
+        };
+
+        /// <summary>
         /// All features enabled
         /// [USE WITH CAUTION] In using this value, you expose your code to future features.
         /// </summary>
-        All = ~0
+        public static Features All => PowerFxV1;
+
+        internal Features()
```

Comment:
>internal [](http://example.com/codeflow?start=8&length=8)

Before, this was public and hosts can set their flags. 
We may need to keep publc.